### ISSUE TITLE
[1.11] shows g_warning on start as new required component 'dock' is empty

### ISSUE THREAD

**Issue by monsta** [mate-desktop/mate-session-manager#106]
This is a follow-up to discussion at https://github.com/mate-desktop/mate-session-manager/commit/6745132519542d1ad81a1dbb91f7fbe9d19c72e7 - I think it's better be a new report.

So, now that we have a new required component named ```'dock'```, m-s-m tries to find the corresponding desktop file for it and fails - as the key is empty:
```
$ gsettings get org.mate.session.required-components dock
''
```
And we get a warning which can be seen in ```~/.xsession-errors```:
```
x-session-manager[584]: WARNING: Unable to find provider '' of required component 'dock'
```
I think that most distros don't ship any docks with MATE by default, so people will see it in 1.12 and wonder what's going on. In fact, I think only Ubuntu MATE users won't see it as the dock is shipped by default there...

One possible solution was suggested by @clefebvre:

> I think it's good that msm warns when it can't find something it's supposed to launch. But if that something is an empty string, then it probably shouldn't warn since it's obviously not missing.. it's just not set.
To summarize, I think we should test if the string is empty.. and if it is, we shouldn't issue any warning.

That would solve it - but we might overlook a real issue when some *really* important required component isn't set. The required components list indeed contains some important pieces you'd not want to run a session *without*:
```
$ gsettings get org.mate.session required-components-list
['windowmanager', 'panel', 'filemanager']
```
It's labeled "required" for a reason as you can see. :smile:
I'm not sure the dock even fits in here...

@clefebvre @flexiondotorg @raveit65 @willysr @posophe @obache
thoughts guys?

**Comment by willysr** [comment]
Will this be enough?
https://github.com/robint99/dock-applet

**Comment by clefebvre** [comment]
I think it's ok @monsta ... default values are set here, so if the string was empty, it would have been set on purpose to be empty.

I really don't see the point in warning the user that <empty> can't be found on the system.

**Comment by monsta** [comment]
> I really don't see the point in warning the user that "empty" can't be found on the system.

Well that's what we see - that g_warning is about empty value...

**Comment by cokeflash** [comment]
How to fix it and add Plank support in org.mate.session.gschema.xml?

I'll try that:

<key name="dock" type="s">
      <default>'['plank']</default>
      <summary>Dock</summary>
      <description>A dock provides a dockable area, similar to a panel, for launching and switching applications.</description>
 </key>

But it don't worked.
g_object_unref: assertion 'G_IS_OBJECT (object)' failed
mate-session[1953]: WARNING: Unable to find provider '' of required component 'dock'

Any idea to fix it?

**Comment by shirishag75** [comment]
As the original reporter of the bug on the Debian side, it very much is there, to date in m-s-m 1.24.0.2 . It hasn't been fixed. Either have a dock or don't have a dock. Don't make the user confused.

**Comment by Cyrille37** [comment]
Hi
Many years later, I come here because I've got the same warning `mai 14 05:44:13 ubuntu-NS5x-NS7xAU mate-session[7677]: WARNING: Unable to find provider '' of required component 'dock'`.
- Linux Mint 21.3
- Mate 1.26
- No dock

The issue is : don't using a dock should not be a warning ;-)

Thanks & cheers

**Comment by zhuyaliang** [comment]
When I have enough time, I will come to deal with this issue
